tests/e2e/flowtest/pvc_snapshot_and_clone/test_pgsql_pvc_snapshot_and_clone_with_base_operation.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import pytest
+
+from concurrent.futures import ThreadPoolExecutor
+
+from ocs_ci.framework.testlib import (
+    skipif_ocs_version, E2ETest, flowtests
+)
+from ocs_ci.ocs.node import drain_nodes, schedule_nodes
+from tests.disruption_helpers import Disruptions
+from ocs_ci.ocs import flowtest
+
+log = logging.getLogger(__name__)
+
+
+@flowtests
+@skipif_ocs_version('<4.6')
+@pytest.mark.polarion_id("OCS-2302")
+class TestPvcSnapshotAndCloneWithBaseOperation(E2ETest):
+    """
+    Tests Story/Flow based test scenario for pgsql snapshot and clone
+    """
+
+    @pytest.fixture(autouse=True)
+    def pgsql_setup(
+        self, pgsql_factory_fixture, snapshot_factory,
+        snapshot_restore_factory, pvc_clone_factory
+    ):
+        # Deploy PGSQL workload
+        log.info("Deploying pgsql workloads")
+        self.pgsql = pgsql_factory_fixture(replicas=3, clients=3, transactions=600)
+
+    def test_pvc_snapshot_and_clone(
+        self, nodes, multiple_snapshot_and_clone_of_postgres_pvc_factory
+    ):
+        """
+        1. Deploy PGSQL workload
+        2. Take a snapshot of the pgsql PVC.
+        3. Create a new PVC out of that snapshot or restore snapshot
+        4. Create a clone of restored snapshot
+        5. Attach a new pgsql pod to it.
+         5. Resize cloned pvc
+        7. Create snapshots of cloned pvc and restore those snapshots
+        8. Attach a new pgsql pod to it and Resize the new restored pvc
+        9. Repeat the above steps in bg when performing base operation:
+            restart pods, worker node reboot, node drain, device replacement
+
+        """
+
+        log.info("Starting multiple creation & clone of postgres PVC in Background")
+        bg_handler = flowtest.BackgroundOps()
+        executor_run_bg_ops = ThreadPoolExecutor(max_workers=1)
+        pgsql_snapshot_and_clone = executor_run_bg_ops.submit(
+            bg_handler.handler,
+            multiple_snapshot_and_clone_of_postgres_pvc_factory,
+            pvc_size_new=25, pgsql=self.pgsql, iterations=1
```

Comment:
will this single iteration of multiple_snapshot_and_clone_of_postgres_pvc_factory runs till the end of the test case?

Reply:
yes as far observed runs till the end of the test.